src/library/scala/Unit.scala

```diff
@@ -23,6 +23,7 @@ final abstract class Unit private extends AnyVal {
   override def getClass(): Class[Unit] = ???
 }
 
+@scala.annotation.compileTimeOnly("not for use in source code. The unit value is written '()'.")
```

Comment:
I wonder what the best way to quote this is. I can imagine someone writing `'()'` and becoming confused. Maybe `"Use () for the value of type Unit"`.

Reply:
We went down that path on the other PR. It sounds like "Write 'Unit()'."

Reply:
Someone who is going to read this is already confused about the term and type name of `Unit`. "not for use in source code." might be confusing since the person might say "what? I can write `def foo(): Unit = Unit`"

Maybe:
```
warn-unit.scala:14: error: `Unit` companion object is not allowed in source; instead, use `()` for the unit value.
    Unit
    ^
```

Reply:
Thanks, you can probably tell that I came up with the "not for use" phrase myself, it wasn't vetted on the other PR. If this PR goes ahead, and passes jenkins somehow, I have a special weekend retreat lined up to reflect upon the sum of commentary.

Reply:
happy to run it by the ministry of quotations for you

Reply:
Can I expense the retreat though them, or is it not that sort of ministry? Or maybe they have a place in Big Sur.